### ISSUE TITLE
Immersive Portals compatibility

### ISSUE THREAD

**Issue by Asteroid4** [nonamecrackers2/crackers-wither-storm-mod#1016]
basically, if immersive portals forge is installed, the head portal is an immersive portal.

**Comment by Asteroid4** [comment]
it might be too laggy tho

**Comment by FrozenDragon0** [comment]
I get the feeling that making the bowels entrance portal look like a immersive portal would be extremely difficult or impossible to do.

**Comment by myron-obsolete-account** [comment]
The dimension-in-dimension effect that Immersive Portals gives is based on the nether portal and end portal blocks, they're basically the gateway. Doing that on an entity would, in the best case, break your game.

**Comment by myron-obsolete-account** [comment]
And here's the worst part: the Wither Storm is rotating and moving. It's impossible, like FrozenDragon0 said, to put that dimension gateway in there (as far as my common sense can guess).

**Comment by Globalfirestorm10** [comment]
> And here's the worst part: the Wither Storm is rotating and moving. It's impossible, like FrozenDragon0 said, to put that dimension gateway in there (as far as my common sense can guess).

Yeah, even if it was possible, it would probably take as much time as it took to make the first release of CWSM to implement this idea (which was roughly 1 year), which (at least for any dev i've seen) is obviously not worth it for a small aesthetic that's only shown when you have another mod

**Comment by nonamecrackers2** [comment]
I doubt this would be possible and would need to be looked into

**Comment by SuperRicky14** [comment]
Immersive portals actually already uses an entity for it's portals, and they already have built in support for moving, animating and rotating portals with commands (just tested this in 1.19.2 immersive portals forge). This pretty much means the wither storm could just create an immersive portals entity with the portals end location set inside the bowels (you can also make portals one way by just deleting the portal entities inside the bowels, but there's probably a better way of doing it), then you could reuse the same code that keeps the wither storm heads stuck to the wither storm, except just put the portal entity at the spot where the purple hole that leads into the bowels is. So it is definitely possible to have this immersive portals compat with cwsm, you just need to figure out how to keep the portals attached to the wither storm and how to set them to the wither storm's bowels, and the mod pretty much handles everything else for you.

**Comment by BluePineappleGuy** [comment]
> then you could reuse the same code that keeps the wither storm heads stuck to the wither storm, except just put the portal entity at the spot where the purple hole that leads into the bowels is. 

Harder than you think, the heads are not separate entities so they don't have a "stuck to the storm" code. Another issue is lag which would cause the portal to lag behind when the storm is moving, making it clip out of place and look very odd.

**Comment by ScreamingMC** [comment]
Our brains think it's impossible. It is possible, but lots of work, mixins...

Things that seem impossible, can be possible if enough work

**Comment by BluePineappleGuy** [comment]
> Things that seem impossible, can be possible if enough work

possible, but likely not worth the effort

**Comment by ScreamingMC** [comment]
Everything can be worth, even a little. Don't call your work that you've put much effort to it as worthless. It shows that it's possible, so don't call your own work like that.
